src/blas/backends/cublas/cublas_helper.hpp

```diff
@@ -180,11 +180,23 @@ class cuda_error : virtual public std::runtime_error {
         throw cublas_error(std::string(#name) + std::string(" : "), err); \
     }
 
-#define CUBLAS_ERROR_FUNC_T(name, func, err, ...)                        \
-    err = func(__VA_ARGS__);                                             \
+#define CUBLAS_ERROR_FUNC_SYNC(name, err, handle, ...)                    \
+    err = name(handle, __VA_ARGS__);                                      \
+    if (err != CUBLAS_STATUS_SUCCESS) {                                   \
+        throw cublas_error(std::string(#name) + std::string(" : "), err); \
+    }                                                                     \
+    cudaStream_t currentStreamId;                                         \
+    CUBLAS_ERROR_FUNC(cublasGetStream, err, handle, &currentStreamId);    \
+    cuStreamSynchronize(currentStreamId);
+
+#define CUBLAS_ERROR_FUNC_T(name, func, err, handle, ...)                \
```

Comment:
Not that important, but I suggest `CUBLAS_ERROR_FUNC_T_SYNC` here (since it's synchronizing), I think it's more consistent with what we had before and what we did in #215.

Reply:
Right, I thought about it and got lazy. I will update all the cublas backend files with new macro name.

Reply:
macro name is updated with `_SYNC`